WF-CDPInstaller/plugin_dir/apache_plugin.py

```diff
@@ -0,0 +1,267 @@
+                if ret != 0:
+                    utils.print_failure()
+                    utils.exit_with_message('a2enmod command was not found')
+                utils.print_success()
+        elif self.os == config.REDHAT:
+            utils.cprint()
+            utils.cprint(
+                'To enable server status page for the apache web,\n'
+                'ensure that mod_status.so module is enabled.\n'
+                'This module is often enabled by default.\n'
+                '"LoadModule status_module modules/mod_status.so"\n'
+                'such line should be included in one of the conf files.\n')
+            _ = utils.cinput('Press Enter to continue.')
+
+        utils.cprint()
+        utils.cprint(
+            'In order to fully utilize the apache plugin with collectd,\n'
+            'the ExtendedStatus setting needs be turned on.\n'
+            'This setting can be turned on by having "ExtendedStatus on"\n'
+            'in one of the .conf file.\n')
+
+        utils.cprint(
+            'If you have already enabled this status, '
+            'answer "no" to the next question.\n'
+            'If you would like us to enable this status, '
+            'answer "yes" and we will '
+            'include a extendedstatus.conf file in your apache folder.\n')
+
+        res = utils.ask(
+            'Would you like us to enable '
+            'the ExtendedStatus?')
+
+        if res:
+            # dir changes depending on the system
+            # tested on Ubuntu 14.04, RHEL 7.2
+            if self.os == config.DEBIAN:
+                conf_dir = '/etc/apache2/conf-enabled'
+                app_name = 'apache2'
+            elif self.os == config.REDHAT:
+                conf_dir = '/etc/httpd/conf.modules.d'
+                app_name = 'httpd'
+
+            utils.print_step('Checking if ' + conf_dir + ' exists.')
+            if utils.check_path_exists(conf_dir):
+                # pull config file here
+                utils.print_success()
+                self.include_apache_es_conf(conf_dir)
+                utils.cprint(
+                    'extendedstatus.conf is now included in the '
+                    '{0} dir.\n'.format(conf_dir))
+                utils.print_step('Restarting apache')
+                ret = utils.call_command(
+                    'service {app_name} restart >> '
+                    '{log} 2>&1'.format(
+                        app_name=app_name,
+                        log=config.INSTALL_LOG))
+                if ret != 0:
+                    raise Exception(
+                        'Failed to restart apache service.')
+                utils.print_success()
+            else:
+                exit_with_message(
+                    '{cond_dir} dir does not exist. Manual '
+                    'set up is required. For help, please '
+                    'consule support@wavefront.com'.format(
+                        conf_dir=conf_dir))
+
+    def write_plugin(self, out):
```

Comment:
If there are failures during the interactive part, say a failed HTTP status or a user Ctrl-C, will a partial and malformed configuration be written?  if so, that doesn't seem good, and can be avoided by separating out questions/detection (data collection) and config file writing using the data.